clients/spark/core/src/main/scala/io/treeverse/gc/UncommittedGarbageCollector.scala

```diff
@@ -52,11 +52,11 @@ object UncommittedGarbageCollector {
     val oldDataPath = new Path(storageNamespace)
     // TODO (niro): implement parallel lister for old repositories (https://github.com/treeverse/lakeFS/issues/4620)
     val oldDataDF = new NaiveDataLister().listData(configMapper, oldDataPath)
-    dataDF.union(oldDataDF)
+    dataDF =
+      dataDF.union(oldDataDF).filter(dataDF("last_modified") < before.getTime).select("address")
 
-    dataDF = dataDF.filter(dataDF("last_modified") < before.getTime).select("address")
     // TODO (niro): Check if really needed and consider passing it as part of uncommittedGCRunInfo
-    dataDF = dataDF.filter(dataDF("address").startsWith("_lakefs"))
+    dataDF = dataDF.filter(!dataDF("address").startsWith("_lakefs"))
```

Comment:
I might be misunderstanding something..
How is `_lakefs\` being listed at all? I thought the `oldDataDF` will only list objects which are directly under the storage namespace. If it lists `_lakefs\` why doesn't it list `data\` too?

Reply:
That's why I added the comment - anyway this is not the only issue here

Reply:
We possibly list everything twice, I think we should address it now (can be in a separate PR though)

Reply:
We don't list everything twice - this is already tested